src/Kubernetes/KubernetesApiExtensions.cs

```diff
@@ -503,12 +503,37 @@ public static V1Job WithBillingId(this V1Job job, string billingId)
             });
         }
 
+        /// <summary>
+        /// Adds a policy failure action and exit codes to the job.
+        /// </summary>
+        /// <param name="job">The job object to modify.</param>
+        /// <param name="actionExitCodeId">A list with all action names and corresponding exit code number.</param>
+        /// <returns>The Kubernetes Job object with added pod failure policy rules.</returns>>
+        public static V1Job WithPodPolicyFailureExitCodes(this V1Job job,
+            Dictionary<ValueTuple<string, List<int>>, ValueType> actions)
+        {
+            job.Spec ??= new V1JobSpec();
+            job.Spec.PodFailurePolicy ??= new V1PodFailurePolicy();
+
+            job.Spec.PodFailurePolicy.Rules = actions.Select(action => new V1PodFailurePolicyRule
```

Comment:
you still do no account for `Rules` being not empty, I think?

Reply:
No I wasn't, nice catch! Now they are concatenating new rules if there are existing ones in [ced6fdc](https://github.com/SneaksAndData/esd-services-sdk/pull/75/commits/ced6fdc3a41523168681d562d976426718063bb6)
Thanks 🙂